trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/signing/InternalSigningContext.java

```diff
@@ -13,51 +13,29 @@
  */
 package io.trino.s3.proxy.server.signing;
 
-import software.amazon.awssdk.auth.signer.params.AwsS3V4SignerParams;
-
 import static java.util.Objects.requireNonNull;
 
 class InternalSigningContext
```

Comment:
If you don't mind... given the refactoring in this PR `InternalSigningContext` separate from `SigningContext` isn't necessary any more. Please make `SigningContext` a record and remove `InternalSigningContext`